src/glmfit.jl

```diff
@@ -731,3 +731,35 @@ function checky(y, d::Binomial)
     end
     return nothing
 end
+
+# need to add :pearson
+const _RESIDUAL_TYPES = [:deviance, :response, :working]
+
+"""
+    residuals(model::GeneralizedLinearModel; type=:deviance)
+
+Return the residuals of a GLM.
+
+Supported values for `type` are:
+- `:deviance` (the default): the signed square root of the element-wise
+  contribution to the deviance
+- `:response`: the difference between the observed and fitted values
+- `:working`: working residuals (used during the IRLS process)
+"""
+function residuals(model::GeneralizedLinearModel; type=:deviance)
+    type in _RESIDUAL_TYPES ||
+        throw(ArgumentError("Unsupported type `$(type)``; supported types are" *
+                            "$(_RESIDUAL_TYPES)"))
+    # TODO: add in optimized method for normal with identity link
+    if type == :response
+        return response(model) - fitted(model)
+    elseif type == :deviance
+        # XXX I think this might be the same as
+        # 2 * wrkresid, but I'm not 100% sure if that holds across families
```

Comment:
@dmbates Does this relationship hold across families?

Reply:
No, it seems that that function is lower level and used in the computation of the associated field rather returning said field.